lib/utils/axios-helpers.js

```diff
@@ -0,0 +1,31 @@
+const axios = require('axios');
+
+const auth = require('../auth');
+const config = require('../config');
+
+/**
+ * Set axios' default auth header to OAuth Bearer Token from v5 RO API
+ *
+ * @async
+ * @returns {Promise} If an error is returned from `auth.getToken` the
+ * promise will reject, otherwise it will resolve.
+ *
+ * @example
+ * await setV5Token()
+ */
+const setV5Token = () =>
+  new Promise((resolve, reject) => {
```

Comment:
Question: I'm not seeing the actual caching mechanism. Should we be checking for a valid (unexpired) `axios.defaults.headers.common.Authorization` (or something) before hitting the promise?

Reply:
One other thing that comes to mind, which may require some follow-up: Eventually — when we continue with the refactor — we'll be using Axios for both V4 and V5. In that case, we'll likely need to manage switching between V4 and V5 auth tokens…

Reply:
Hey Shane, good question! Rather duplicating logic, I am allowing `auth.getToken` to do the caching/token refresh (if needed). 

lib/auth.js:228
`  
} else if (auth.token && auth.token.access_token && auth.token.expires && auth.token.expires > new Date()) {
    /*
     * If there's already a valid token, use it
     */
    callback(null, auth.token.access_token);
`

Since it is reading from memory, I don't think there are any performance concerns with using the caching mechanism in the `auth` module. If you disagree, we can add logic to `setV5Token`.

To you second comment, from the comments of the [group channel](https://rewardops.slack.com/archives/G019XSSBPJT/p1600883349008000) I think we need to refactor auth module to cache v4 and v5 token separately, since they are not interchangeable.

Reply:
> To you second comment, from the comments of the group channel I think we need to refactor auth module to cache v4 and v5 token separately, since they are not interchangeable.

IIRC, now that I've been refreshed on the details, the tokens _are_ interchangeable, but a non-PII token cannot be used with a PII-enabled program making PII calls.

Reply:
So what we could do is: We rename the `setV5Token` to `setPiiToken` and, any time a program is PII-enabled, we use `setPiiToken`; otherwise, we can use the standard token process